src/BenchmarkDotNet/Parameters/ParameterComparer.cs

```diff
@@ -24,13 +24,35 @@ public int Compare(ParameterInstances x, ParameterInstances y)
             if (x == null) return -1;
             for (int i = 0; i < Math.Min(x.Count, y.Count); i++)
             {
+                //todo: compare non-primitive types too
                 int compareTo = PrimitiveComparer.CompareTo(x[i]?.Value, y[i]?.Value);
                 if (compareTo != 0)
                     return compareTo;
+
+                int arrayCompareTo = CompareArrays(x[i]?.Value, y[i]?.Value);
```

Comment:
I was surprised to see that it works, then I've checked and found that we are relying on the `.ToString` comparison:

https://github.com/dotnet/BenchmarkDotNet/blob/58d4bae8996b713f21d6472702d95b21a77f79c9/src/BenchmarkDotNet/Parameters/ParameterComparer.cs#L49

It would be more future proof to perform an explicit check: 

```cs
int compareTo = x[i]?.Value is Array xArray && y[i]?.Value is Array yArray
   ? CompareArrays(xArra, yArray)
   : PrimitiveComparer.CompareTo(x[i]?.Value, y[i]?.Value);
```

Reply:
But what the code will do if the type is neither an array nor a primitive? :)

The code works like this:
```
1) if type is primitive:
1.1) if (x > y or x < y) : return 1 or -1
2) if type is array:
1.1) if (x > y or x < y) : return 1 or -1
3) compare by string and return
```

So, if the `x=0 and y=0`, it exits on the last line:
```
return string.CompareOrdinal(0?.ToString(), 0?.ToString()); 
```